### ISSUE TITLE
Error saving assigned

### ISSUE THREAD

**Issue by vitormarques86** [glpi-project/glpi#11796]
### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Version

10.0.1

### Bug description

After assigning a ticket to a user, when I try to save, the screen goes blank.

### Relevant log output

```shell
Uncaught Exception TypeError: Argument 1 passed to Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier() must be of the type string, array given, called in /var/www/html/glpi/src/DBmysql.php on line 1257 in /var/www/html/glpi/src/Toolbox/Sanitizer.php at line 228
```

**Comment by cedric-anne** [comment]
Hi,

Have you some active plugins ? In this case, try to deactivate them (especially `behaviours` plugin) to see if problem persist.

**Comment by cconard96** [comment]
I see the same error when adding any actors to tickets.
I haven't been able to track down the cause completely yet, but it doesn't seem related to plugins and seemed to have started with a commit merged yesterday or today.

**Comment by vitormarques86** [comment]
Even with the plugins disabled, the problem persists.

**Comment by vitormarques86** [comment]
Happened after upgrading to version 10.0.1

**Comment by brunnersm** [comment]
Me too, after upgrading to version 10.0.1
[2022-06-02 15:46:57] glpiphplog.CRITICAL:   *** Uncaught Exception TypeError: Argument 1 passed to Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier() must be of the type string, array given, called in /var/www/html/src/DBmysql.php on line 1257 in /var/www/html/src/Toolbox/Sanitizer.php at line 228
  Backtrace :
  src/DBmysql.php:1257                               Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier()
  src/DBmysql.php:1288                               DBmysql::quoteValue()
  src/DBmysql.php:1312                               DBmysql->buildInsert()
  src/CommonDBTM.php:705                             DBmysql->insert()
  src/CommonDBTM.php:1297                            CommonDBTM->addToDB()
  src/CommonITILObject.php:3006                      CommonDBTM->add()
  src/CommonITILObject.php:1716                      CommonITILObject->addAdditionalActors()
  src/Ticket.php:1303                                CommonITILObject->prepareInputForUpdate()
  src/CommonDBTM.php:1565                            Ticket->prepareInputForUpdate()
  front/ticket.form.php:84                           CommonDBTM->update()

**Comment by ariasfonseca** [comment]
Hi, me too  not technical aggregation for ticket. event log error:

[2022-06-02 16:11:18] glpiphplog.CRITICAL:   *** Uncaught Exception TypeError: Argument 1 passed to Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier() must be of the type string, array given, called in /var/www/html/glpi/src/DBmysql.php on line 1257 in /var/www/html/glpi/src/Toolbox/Sanitizer.php at line 228
  Backtrace :
  src/DBmysql.php:1257                               Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier()
  src/DBmysql.php:1288                               DBmysql::quoteValue()
  src/DBmysql.php:1312                               DBmysql->buildInsert()
  src/CommonDBTM.php:705                             DBmysql->insert()
  src/CommonDBTM.php:1297                            CommonDBTM->addToDB()
  src/CommonITILObject.php:2987                      CommonDBTM->add()
  src/CommonITILObject.php:1716                      CommonITILObject->addAdditionalActors()
  src/Ticket.php:1303                                CommonITILObject->prepareInputForUpdate()
  src/CommonDBTM.php:1565                            Ticket->prepareInputForUpdate()
  front/ticket.form.php:84                           CommonDBTM->update()

**Comment by OlegusPF** [comment]
Hi
We resolved this problem. I don.t know is it fully correct or not, but it is work :)
You need to change 228 string to
....  isNsClassOrCallableIdentifier($value)  ....
and change the string 231 to
...  , (string)$value, ...

**Comment by cedric-anne** [comment]
> Hi We resolved this problem. I don.t know is it fully correct or not, but it is work :) You need to change in the file /glpi/src/Toolbox/Sanitizer.php 228 string to .... isNsClassOrCallableIdentifier($value) .... and change the string 231 to ... , (string)$value, ...

Well, you just hide the problem, and it will probably add corrupted data into your database, so it is not a good idea.

**Comment by OlegusPF** [comment]
ok, thanks, i was revert it. waiting for resolve )

**Comment by dani** [comment]
I have the same issue on one instance of GLPI, but on another instance, I can't reproduce. Both are almost similar. Have been upgraded from 10.0.0 to 10.0.1 today, running on AlmaLinux with the same version of PHP (8.0.19) and almost the same MariaDB (10.6.7 vs 10.6.8). I'm trying to figure out the difference which make the problem reproducible.

**Comment by wevrard** [comment]
in my fresh update from 10.0.1, i have a same error. 

[2022-06-03 16:12:02] glpiphplog.CRITICAL:   *** Uncaught Exception TypeError: Argument 1 passed to Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier() must be of the type string, array given, called in /var/www/html/glpi/src/DBmysql.php on line 1257 in /var/www/html/glpi/src/Toolbox/Sanitizer.php at line 228
  Backtrace :
  src/DBmysql.php:1257                               Glpi\Toolbox\Sanitizer::isNsClassOrCallableIdentifier()
  src/DBmysql.php:1288                               DBmysql::quoteValue()
  src/DBmysql.php:1312                               DBmysql->buildInsert()
  src/CommonDBTM.php:705                             DBmysql->insert()
  src/CommonDBTM.php:1297                            CommonDBTM->addToDB()
  src/CommonITILObject.php:3006                      CommonDBTM->add()
  src/CommonITILObject.php:1716                      CommonITILObject->addAdditionalActors()
  src/Ticket.php:1303                                CommonITILObject->prepareInputForUpdate()
  src/CommonDBTM.php:1565                            Ticket->prepareInputForUpdate()
  front/ticket.form.php:84                           CommonDBTM->update()

**Comment by cconard96** [comment]
> I'm trying to figure out the difference which make the problem reproducible.

Do you have any Business Rules for Tickets on the non-working instance? Are they triggered by Add, Update or both?

**Comment by dani** [comment]
> > I'm trying to figure out the difference which make the problem reproducible.
> 
> Do you have any Business Rules for Tickets on the non-working instance? Are they triggered by Add, Update or both?

Indeed, the "broken" instance has business rules (Add / Add+Update), while the working instance has no business rule at all

**Comment by cosmedd** [comment]
We have the same problem with GLPI 10.0.1

It didn't happen with GLPI 10.0 until commit 3de7b01cfd202ca497aecdf6735e2edaa1e87a67

We can confirm that the problem persists with all plugins disabled.

Only happens when assigning or changing the status of already created tickets. It is not reproduced in new tickets after upgrading to GPI 10.0.1.

**Comment by cconard96** [comment]
Issue started with the commit from #11728 but it does not seem to be the cause directly. The code that seems to be the issue is old so I don't know why the issue hasn't presented before now.

The issue would only affect updates from the form (not massive actions) because of the different way it handles actors.

**Comment by dani** [comment]
I can confirm that adding a single business rule (triggered on Add/Update) is enough to make my working instance broken too

**Comment by cosmedd** [comment]
Hello, for us it seems that the problem starts with this commit: 5de1bac5e390cb1f200a299eadf7d2d9d58cd3f9

**Comment by cosmedd** [comment]
Regardless of whether or not it is safe to reverse it at the very least I can confirm that it is not done cleanly. That is, it would be necessary to revert more commits or perform a manual cleanup.

**Comment by ariasfonseca** [comment]
thanks @brunnersm this solution work for  me

**Comment by brunoproza** [comment]
@brunnersm this solution work for me too, Tks

**Comment by vitormarques86** [comment]
@brunnersm tks!

**Comment by cmoreno038** [comment]
I only found this difference
![imagen](https://user-images.githubusercontent.com/44004531/171917611-facce09b-38a2-491f-91f3-572584c05f4f.png)

**Comment by DooBio** [comment]
Hello

I'm sorry but the fix doesn't work for me.
I am in 10.0.1 and I have modified line 130 to 148 the file src/RuleTicketCollection.php

`// append _actors values to awaited indexes by rule engine
        if (isset($input['_actors']) && count($input['_actors'])) {
            foreach (['requester', 'observer', 'assign'] as $actortype) {
                if (isset($input['_actors'][$actortype]) && count($input['_actors'][$actortype])) {
                    foreach ($input['_actors'][$actortype] as $actor) {
                        $input_key = "_" .
                                    getForeignKeyFieldForItemType($actor['itemtype']) .
                                    "_" .
                                    $actortype;

                        if (!isset($input[$input_key])) {
                            $input[$input_key] = [];
                        }

                        $input[$input_key][] = $actor['items_id'];
                    }
                }
            }
        }

`

In my ticket when I want to add an actor and I save I come across this blank page "http://*.*.*.*/glpi/front/ticket.form.php?id=2022002385"

and I confirm via the Action button I have no assignment problem

Thanks for your help

**Comment by DooBio** [comment]
After applying : [https://github.com/glpi-project/glpi/pull/11812](url)

I confirm it works

Thx 😉

**Comment by mulbzh** [comment]
I don't find where is the solution in https://github.com/glpi-project/glpi/pull/11812, what is the code to change ?

**Comment by mulbzh** [comment]
Ok, i found the code to change, there is a way to download all the file with the new code ?

**Comment by DamBn** [comment]
Hello 

I don't find where is the solution in #11812 what is the code to change ?
/src/ticket.php is no fonctional

**Comment by paulopera** [comment]
as @DooBio mentioned above here's the [direct commit link ](https://github.com/glpi-project/glpi/pull/11812/commits/30cf046f305b5dc8a5230221a524cf6ef465f142)

**Comment by DamBn** [comment]
@paulopera I replaced the ticket.php file but nothing changes. I'm using PHP 8.0.5 does this have impact ?

**Comment by variamus** [comment]
> I copied 10.0.0/src/RuleTicketcCollection.php to production and it fixed the problem, until the fix came.

It's working. However, is it safe to do ? No database corruption or anything else ?
Thanks

**Comment by cedric-anne** [comment]
Hi,

It should be fixed by #11957.
